### PR TITLE
Finished DM Feature

### PR DESCRIPTION
**Pull request by andreayang310**
I wasn't sure if I'm supposed to add some tests too (and what's a good way of going about that). Also I followed the guide which makes all the DMs public on the user's page, though that might not be the final behavior we want.

### CODE REVIEW

File: src/main/webapp/js/user-page-loader.js [hunk 1]
```diff
@@ -48,18 +48,18 @@ function fetchAboutMe(){
...
       .then((response) => {
         return response.json();
       })
       .then((loginStatus) => {
-        if (loginStatus.isLoggedIn &&
-            loginStatus.username == parameterUsername) {
+        if (loginStatus.isLoggedIn) {
           const messageForm = document.getElementById('message-form');
-          messageForm.classList.remove('hidden');
+        messageForm.action = '/messages?recipient=' + parameterUsername;
```
Comment (ctrokid) [inline src/main/webapp/js/user-page-loader.js]: Hmmm. Is this formatted correctly? If it's not, I have no clue why the linter did not catch this 😆

File: src/main/webapp/js/user-page-loader.js [hunk 2]
```diff
@@ -48,18 +48,18 @@ function fetchAboutMe(){
...
+        messageForm.classList.remove('hidden');
```
Comment (ctrokid) [inline src/main/webapp/js/user-page-loader.js]: Same small thing about formatting.

**Review by ctrokid — CHANGES_REQUESTED** [review]
Small formatting, then this is good to go.

File: src/main/webapp/js/user-page-loader.js (hunk 2 above)
Comment (andreayang310) [reply src/main/webapp/js/user-page-loader.js]: The linter didn't catch it but I just fixed it by hand!

**Review by ctrokid — APPROVED** [review]

**Pull request merged**